packages/fury-adapter-oas3-parser/lib/parser/oas/parseServerObject.js

```diff
@@ -15,6 +15,38 @@ const parseServerVariableObject = require('./parseServerVariableObject');
 const name = 'Server Object';
 const requiredKeys = ['url'];
 
+const validateVariableInURL = (context, object) => {
+  const url = object.getValue('url');
+  const variables = object.get('variables');
+  const parseResult = new context.namespace.elements.ParseResult();
+
+  const variablesKeys = variables.content.map(variable => variable.key.toValue());
+  const urlVariables = url
+    .match(/{(.*?)}/g)
+    .map(x => x.replace(/[{}]/g, ''));
+
+  // if you define a variable that is not in URL it warns (and the variable is discarded).
+  variablesKeys.forEach((key) => {
+    if (urlVariables.indexOf(key) === -1) {
+      parseResult.push(createWarning(context.namespace,
+        `Server variable '${key}' is not present in the URL and will be discarted`, variables));
+
+      variables.remove(key);
+    }
+  });
+
+  // if you place a variable in the URL and its not in variables you get a warning that the variable is missing.
+  urlVariables.forEach((key) => {
+    if (!variables.hasKey(key)) {
+      parseResult.push(createWarning(context.namespace,
+        `URL variable '${key}' is missing within the server variables`, variables));
```

Comment:
```suggestion
        `URL variable '${key}' is missing within the server variables`, object.get('url')));
```

I think we want to use the URL here to create the warning, because the source map for the warning should originate from the URL which contains the variable which isn't defined.

Reply:
i see, changed